### PR TITLE
Add the DeviceType proto to Auth Service

### PR DESCRIPTION
**Pull request by codingllama**
Replaces the local device type in AddMFADeviceRequestInit for a global enum.
Useful for future RPCs.

### CODE REVIEW

**Comment by codingllama** [comment]
Follow up from @kimlisa's comment in https://github.com/gravitational/teleport/pull/8334 (branched from it as well). Once https://github.com/gravitational/teleport/pull/8334 is in, I'll change target to master.

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -1772,10 +1772,22 @@ func addMFADeviceRegisterChallenge(gctx *grpcContext, stream proto.AuthService_A
 	// RPCs.
 	webIdentity := wanlib.WithInMemorySessionData(auth.Identity)
 
+	devType := initReq.DeviceType
+	if devType == proto.DeviceType_DEVICE_TYPE_UNSPECIFIED {
+		// Try and convert from legacy type.
+		// Keep conversion until 9.x, when the field is marked for deletion.
+		m := map[proto.AddMFADeviceRequestInit_LegacyDeviceType]proto.DeviceType{
```
Comment (zmb3) [inline lib/auth/grpcserver.go:1779]: I would mark this with `DELETE IN` so we can easily grep for it.

**Review by zmb3 — APPROVED** [review]
Just a minor nit but looks good.

File: lib/auth/grpcserver.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/grpcserver.go:1779]: Once you delete the proto enum this one falls as a consequence, so I didn't mark it.

**Review by klizhentas — APPROVED** [review]

**Comment by codingllama** [comment]
Many thanks for the fast reviews, @klizhentas.

**Review by r0mant — APPROVED** [review]

**Pull request merged**